pulser-core/pulser/backend/remote.py

```diff
@@ -17,7 +17,10 @@
 import typing
 from abc import ABC, abstractmethod
 from enum import Enum, auto
-from typing import Any, TypedDict
+from types import TracebackType
+from typing import Any, Type, TypedDict, cast
+
+from typing_extensions import Self
```

Comment:
Hum, this might be problematic, `typing_extensions` is not in our requirements

Reply:
ah so would it be an issue for me to introduce the dependency?
Linting fails if I don't type `Self` and I can't type `Self` for 3.8 without it

Reply:
I would say this doesn't warrant the extra dependency since we have alternatives:

1. We might not even be returning `Self` in the end (see comments below)
2. You can use a `TypeVar` instead, as mentioned [here](https://docs.python.org/3/library/typing.html#typing.Self)